app/src/main/java/fr/free/nrw/commons/upload/SpinnerLanguagesAdapter.java

```diff
@@ -139,12 +141,12 @@ public void init(int position, boolean isDropDownView, String savedLanguageValue
             String languageCode = LangCodeUtils.fixLanguageCode(languageCodesList.get(position));
             final String languageName = StringUtils.capitalize(languageNamesList.get(position));
 
-            if(savedLanguageValue.equals("")){
+            if(savedLanguageValue!=null && savedLanguageValue.equals("")){
```

Comment:
Use TextUtils.isEmpty() instead of two checks

Reply:
done